### PR TITLE
feat(components): adds custom counter value

### PR DESCRIPTION
**Pull request by sdwvit**
... which defaults to children.length for TreeView

**Please check if the PR fulfills these requirements**

- [x] The commit(s) message(s) follows our
  [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**What is the current behavior?** (You can also link to an open issue here)
Counter shows children.length and has no customization opts

**What is the new behavior?**
Counter's value may be customized

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration
path for existing applications: ...

**Other information**:

### CODE REVIEW

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://274.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://274.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://274.talend.surge.sh)

File: packages/components/src/TreeView/TreeViewItem/TreeViewItem.component.js [hunk 1]
```diff
@@ -119,6 +120,7 @@ TreeViewItem.propTypes = {
 			label: PT.string,
 			icon: PT.string,
 		})),
+		counter: PT.number,
```
Comment (talend-glorieux) [inline packages/components/src/TreeView/TreeViewItem/TreeViewItem.component.js]: PT?

Comment (sdwvit) [inline packages/components/src/TreeView/TreeViewItem/TreeViewItem.component.js]: import { PropTypes as PT } from 'react'

Comment (sdwvit) [inline packages/components/src/TreeView/TreeViewItem/TreeViewItem.component.js]: readability ftw

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://274.talend.surge.sh)

File: packages/components/src/TreeView/TreeViewItem/TreeViewItem.component.js (hunk 1 above)
Comment (talend-glorieux) [inline packages/components/src/TreeView/TreeViewItem/TreeViewItem.component.js]: Or not. 
No one in the Talend/ui repository uses PT as PropTypes.
Please align with the common style.

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://274.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://274.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://274.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://274.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://274.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://274.talend.surge.sh)

**Pull request merged**